tests_end_to_end/page_objects/ExperimentItemsPage.py

```diff
@@ -0,0 +1,53 @@
+from playwright.sync_api import Page, expect, Locator
+import re
+
+class ExperimentItemsPage:
+
+    def __init__(self, page: Page):
+        self.page = page
+        self.next_page_button_locator = self.page.locator("div:has(> button:nth-of-type(4))").locator('button:nth-of-type(3)')
+
+    def get_pagination_button(self) -> Locator:
+        return self.page.get_by_role('button', name='Showing')
+
+    def get_total_number_of_items_in_experiment(self):
+        pagination_button_text = self.get_pagination_button().inner_text()
+        match = re.search(r'of (\d+)', pagination_button_text)
+        if match:
+            return int(match.group(1))
+        else:
+            return 0
+        
+    def get_id_of_nth_experiment_item(self, n: int):
+        row = self.page.locator('tr').nth(n+1)
+        cell = row.locator('td').first
+        cell.hover()
+        cell.get_by_role('button').nth(1).click()
+        id = self.page.evaluate('navigator.clipboard.readText()')
+        return id
+    
+    
+    def get_all_item_ids_on_current_page(self):
+        ids = []
+        rows = self.page.locator('tr').all()
+        for row_index, row in enumerate(rows[2:]):
+            item = {}
+            cells = row.locator('td').all()
+            cell = row.locator('td').first
+            cell.hover()
+            cell.get_by_role('button').nth(1).click()
+            id = self.page.evaluate('navigator.clipboard.readText()')
+            ids.append(id)
+
+        return ids
+
+
+    def get_all_item_ids_in_experiment(self):
+        ids = []
+        ids.extend(self.get_all_item_ids_on_current_page())
+        while self.next_page_button_locator.is_visible() and self.next_page_button_locator.is_enabled():
+            self.next_page_button_locator.click()
+            self.page.wait_for_timeout(500)
+            ids.extend(self.get_all_dataset_items_on_current_page())
+
+        return ids
```

Comment:
Minor: mentioned before, but please add a pre-commit hook to format the Python code within this `tests_end_to_end` folder, so among other things, it adds an empty line at the end of each file. Use the SDK linting as reference.

Reply:
good point, i went ahead and added full linting functionality, along with a workflow file to check on every PR. applied it as part of this PR, everything looks good. will add more bits in the future when doing improvements (i.e. clean typing on everything like the SDK code currently has)